bundles/framework/featuredata/instance.js

```diff
@@ -81,6 +86,45 @@ Oskari.clazz.define('Oskari.mapframework.bundle.featuredata.FeatureDataBundleIns
             this.mapModule.registerPlugin(this.plugin);
             this.mapModule.startPlugin(this.plugin);
         },
+
+        /**
+         * Fetches reference to the map layer service
+         * @return {Oskari.mapframework.service.MapLayerService}
+         */
+        getLayerService: function () {
+            return this.sandbox.getService('Oskari.mapframework.service.MapLayerService');
+        },
+        /**
+         * Adds tools for all layers
+         */
+        __setupLayerTools: function () {
+            // add tools for feature data layers
+            this.getLayerService().getAllLayers().filter((layer) => layer.hasFeatureData()).forEach((layer) => this.__addTool(layer, true));
+            // update all layers at once since we suppressed individual events
+            const event = Oskari.eventBuilder('MapLayerEvent')(null, 'tool');
+            this.sandbox.notifyAll(event);
+        },
+        /**
+         * Adds the Feature data tool for layer
+         * @param  {Object} layer layer to process
+         * @param  {Boolean} suppressEvent true to not send event about updated layer (optional)
+         */
+        __addTool: function (layer, suppressEvent) {
+            if (!layer || !layer.hasFeatureData()) {
+                return;
+            }
+            // add feature data tool for layer
+            const tool = Oskari.clazz.create('Oskari.mapframework.domain.Tool');
+            const label = Oskari.getMsg(FEATUREDATA_BUNDLE_ID, 'layer.featureData');
+
+            tool.setName('featureData');
+            tool.setIconCls('show-featuredata-tool');
+            tool.setTitle(label);
+            tool.setTooltip(label);
+            tool.setCallback(() => this.sandbox.postRequestByName('Featuredata.ShowFeatureDataRequest', [layer.getId()]));
```

Comment:
We don't really need to use a request here since we can call featuredata bundles own function in this callback, but the request might come in handy in the future for opening the table programmatically through RPC.